xml_converter/src/xml_converter.cpp

```diff
@@ -138,16 +138,20 @@ void write_taco_directory(
 void write_burrito_directory(
     string output_path,
     map<string, Category>* marker_categories,
-    vector<Parseable*>* parsed_pois) {
+    vector<Parseable*>* parsed_pois,
+    bool split_proto_by_category) {
     if (!filesystem::is_directory(output_path)) {
         if (!filesystem::create_directory(output_path)) {
             cout << "Error: " << output_path << "is not a valid directory path" << endl;
             return;
         }
     }
-    StringHierarchy category_filter;
-    category_filter.add_path({}, true);
-    write_protobuf_file(output_path, category_filter, marker_categories, parsed_pois);
+    if (split_proto_by_category) {
+        write_protobuf_file_per_category(output_path, marker_categories, parsed_pois);
+    }
+    else {
+        write_protobuf_file(output_path, marker_categories, parsed_pois);
+    }
```

Comment:
This logic should probably be nearby the logic for `split by map id`. It is convoluted that split by map id is in the parent function but split by category is in this child function.

Reply:
I have actually considered moving map_id into this function but it would require reworking some other variables so thought it outside this PR scope.